src/Api/WebSocket.hs

```diff
@@ -79,7 +81,12 @@ getDatumsByHashes hashes = do
     Left err ->
       Left $ mkGetDatumsByHashesFault $ Text.pack $ show err
     Right datums ->
-      Right $ mkGetDatumsByHashesResponse (Just $ Aeson.toJSON <$> [datums])
+      let encodedDatums :: Aeson.Value
+          encodedDatums =
+            if Map.null datums
+              then Aeson.toJSONList @(Map.Map DataHash (Either DatabaseError Data)) []
```

Comment:
if we leave `[datum]` as before, it would encode it in the JSON as `[{}]` instead of `[ ]`